101-automation-configuration/azuredeploy.json

```diff
@@ -0,0 +1,301 @@
+    },
+    "virtualNetworkName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the Virtual Network to Create"
+      },
+      "defaultValue": "network"
+    },
+    "virtualNetworkAddressRange": {
+      "type": "string",
+      "metadata": {
+        "description": "The address range of the new virtual network "
+      },
+      "defaultValue": "10.0.0.0/16"
+    },
+    "virtualNetworkSubnet": {
+      "type": "string",
+      "metadata": {
+        "description": "The address range of the subnet created in the new virtual network"
+      },
+      "defaultValue": "10.0.0.0/24"
+    },
+    "nicIPAddress": {
+      "type": "string",
+      "metadata": {
+        "description": "The IP address of the new AD VM"
+      },
+      "defaultValue": "10.0.0.4"
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the Administrator of the new VM and Domain"
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The password for the Administrator account of the new VM and Domain"
+      }
+    },
+    "safeModePassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The password for the safe mode restore feature in the new domain"
+      }
+    },
+    "VMSize": {
+      "type": "string",
+      "metadata": {
+        "description": "The size of the VM Created"
+      },
+      "defaultValue": "Standard_DS2_V2"
+    },
+    "domainName": {
+      "type": "string",
+      "metadata": {
+        "description": "The full qualified domain name to be created"
+      },
+      "defaultValue": "contoso.local"
+    }
+  },
+  "variables": {
+    "automationAccountName": "[concat(parameters('prefix'), 'configuration')]",
+    "publicIPAddressName": "publicIP",
+    "publicIPAddressType": "Dynamic",
+    "virtualNetworkSubnetName": "Subnet",
+    "virtualNetworkSubnets": [
+      {
+        "name": "[variables('virtualNetworkSubnetName')]",
+        "properties": {
+          "addressPrefix": "[parameters('virtualNetworkSubnet')]"
+        }
+      }
+    ],
+    "VMName": "server",
+    "DataDisk": "DataDisk",
+    "DataDiskSize": 1000,
+    "nicName": "nic",
+    "virtual networkID": "[resourceId('Microsoft.Network/virtualNetworks', parameters('virtualNetworkName'))]",
+    "nicSubnetRef": "[concat(variables('virtual networkID'),'/subnets/',variables('virtualNetworkSubnetName'))]",
+    "PublicIpResourceId": "[resourceId(concat('Microsoft.Network','/','publicIPAddresses'),variables('publicIPAddressName'))]",
+    "NicId": "[resourceId('Microsoft.Network/networkInterfaces',variables('nicName'))]",
+    "IPConfigID": "[concat(variables('NicId'),'/ipConfigurations/ipconfig')]",
+    "nestedTemplateBaseURL": "https://raw.githubusercontent.com/mgreenegit/azure-quickstart-templates/101-automation-configuration/101-automation-configuration/nested/",
```

Comment:
See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#samples-that-contain-extra-artifacts-custom-scripts-nested-templates-etc) for how to handle artifacts

Reply:
I don't understand the question here.  This is following exactly what the BP guide recommends by setting a base URL and using that variable in all the URI references, or do I not understand?

Reply:
See the samples in the guide at the link above - you need two parameters as defined.  Then those parameters will be part of each url you need to construct in the template.  To see it working in practices use the scripts in the root of the repo to test deployment.

Reply:
Addressed in new commit https://github.com/Azure/azure-quickstart-templates/pull/4063/commits/6713e116ceb36e11f48345a9b1f96ac5eacb641b